### PR TITLE
Update StyleModel.php

### PR DESCRIPTION
**Pull request by BrainforgeUK**
Corrected language .ini file handling in child templates.

Pull Request for Issue #40660 .

### Summary of Changes

Corected loading of core and/or local language file(s).

### Testing Instructions

Install the following very simple child templates (test1, test2(:

[tpl_cassiopeia_test1.zip](https://github.com/joomla/joomla-cms/files/11581362/tpl_cassiopeia_test1.zip)

[tpl_cassiopeia_test2.zip](https://github.com/joomla/joomla-cms/files/11581363/tpl_cassiopeia_test2.zip)

### Actual result BEFORE applying this Pull Request

actual-screenshot

### Expected result AFTER applying this Pull Request

expected-screenshot

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [*] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [*] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by ChristineWk** [comment]
@brainforgeUK 

got this message:

screen shot 2023-05-27 at 10 07 22

but template folder is writeable. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40665">issues.joomla.org/tracker/joomla-cms/40665</a>.</sub>

**Comment by BrainforgeUK** [comment]
A benign issue, no media files in my test child templates! See ...
https://issues.joomla.org/tracker/joomla-cms/40666

**Comment by BrainforgeUK** [comment]
Additional pull request created to fix benign message.
https://github.com/joomla/joomla-cms/pull/40667

**Comment by brianteeman** [comment]
Seems very odd to me that this is needed as I have had no problems at all with child templates and language strings.

image

You can try any of the zip files at https://github.com/brianteeman/joomlachild/tree/main/examples they are all child templates of cassiopeia

File: administrator/components/com_templates/src/Model/StyleModel.php [hunk 1]
```diff
@@ -396,9 +396,12 @@ protected function preprocessForm(Form $form, $data, $group = 'content')
         $formFile = Path::clean($client->path . '/templates/' . $template . '/templateDetails.xml');
 
         // Load the core and/or local language file(s).
+	    // Default to using parent template language constants
+        $lang->load('tpl_' . $data->parent, $client->path)
+            || $lang->load('tpl_' . $data->parent, $client->path . '/templates/' . $data->parent);
+
+	    // Apply any, optional, overrides for child template language constants
         $lang->load('tpl_' . $template, $client->path)
```
Comment (richard67) [inline administrator/components/com_templates/src/Model/StyleModel.php]:
```suggestion
        // Load the core and/or local language file(s).
        // Default to using parent template language constants
        $lang->load('tpl_' . $data->parent, $client->path)
            || $lang->load('tpl_' . $data->parent, $client->path . '/templates/' . $data->parent);

        // Apply any, optional, overrides for child template language constants
        $lang->load('tpl_' . $template, $client->path)
```
Fix PHP code style errors.

**Comment by BrainforgeUK** [comment]
Is this requiring any action on my part?

**Comment by obuisard** [comment]
> Is this requiring any action on my part?

No, thanks for the PR Jonathan @BrainforgeUK, we now need people to test :-)

**Comment by robinclapp** [comment]
I have tested this item :white_check_mark: successfully on a174660e263175b6ac0b265ee7c0e7031a1f86ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40665">issues.joomla.org/tracker/joomla-cms/40665</a>.</sub>

**Comment by blanderson22** [comment]
I have tested this item :white_check_mark: successfully on a174660e263175b6ac0b265ee7c0e7031a1f86ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40665">issues.joomla.org/tracker/joomla-cms/40665</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40665">issues.joomla.org/tracker/joomla-cms/40665</a>.</sub>

**Pull request merged**